News Panel

DarvishKamalia opened the pull request.

DarvishKamalia committed:
Re-added files

DarvishKamalia committed:
- Added correct versions of News.cpp and NewsPanel.
- Also modified CMakeLists to include News source files and Qt networking
  libraries.
- :)

DarvishKamalia committed:
- Added a new custom widget to display the news items
- Expanded XML parsing to also include description of the links
- Changed the NewsPanel layout to a columnView

DarvishKamalia committed:
modified newsItem layout

DarvishKamalia committed:
Merge branch 'dev' of github.com:HorizonLauncher/Client into dev

DarvishKamalia committed:
- Modified CMakeLists

DarvishKamalia committed:
- Modified layout of NewsWidget to only show headlines
- Added randomization to the feed so that headlines are spread across
  multiple columns
- Cleaned up unnecessary functions
- Cleaned up whitespace
- Changed PCGamer function names to RPS

DarvishKamalia committed:
removed dependencies to OpenSSL libs

DarvishKamalia committed:
I'm stupid

DarvishKamalia commented:
I'm stupid pls ignore.

DarvishKamalia committed:
- Fixed reddit parser
- Fixed bug that put a label above the first column

thomassross commented:
Hi,

Just a couple of things I'd like to be changed:
1. News.cpp: instead of having three different `onFetchComplete<source>` functions, can you put it into one general one? They all do effectively the same thing, so this shouldn't be an issue.
2. Your code isn't fully in line with our [style guide](https://github.com/HorizonLauncher/Client/blob/dev/CONTRIBUTING.md) (Some braces are at the EOL, some extra newlines)
3.  Can you programmatically create the news widget instead of using a UI file?

Thanks.

DarvishKamalia committed:
- Removed redundant RSS parsing functions
- Removed unused html parsing functions

DarvishKamalia committed:
fixed style issues

DarvishKamalia committed:
Modified destructor

DarvishKamalia committed:
Merge branch 'dev' of github.com:HorizonLauncher/Client into dev

- Fixed conflict in CMakeLists.txt on my side
- More style fixes in News.cpp

DarvishKamalia committed:
Fixed build issue

DarvishKamalia committed:
Merge branch 'dev' of github.com:HorizonLauncher/Client into dev

- Fixed conflict in MainPanel.cpp

DarvishKamalia committed:
- Added new window to choose news feeds

DarvishKamalia committed:
Merge branch 'dev' of github.com:HorizonLauncher/Client into dev
Additional changes

DarvishKamalia committed:
- Fixed load from settings file
- Changed NewsPanel to load urls specified by user
- Changed CMakeLists.txt to include debug flag

DarvishKamalia committed:
Merge branch 'dev' of github.com:HorizonLauncher/Client into dev

DarvishKamalia committed:
- Fixed merge conflict stuff
- Added manage news button to setting

DarvishKamalia committed:
Travis are you happy now

DarvishKamalia committed:
pls

DarvishKamalia committed:
- Pulled changes to Settings files from dev
- All three columns in News now have the same number of headlines

DarvishKamalia committed:
Removed debug statement

DarvishKamalia committed:
Copied settings files again

DarvishKamalia committed:
Added news stuff back in

DarvishKamalia committed:
Connected manage news to manage news button

DarvishKamalia committed:
Merge remote-tracking branch 'origin/dev' into dev

thomassross commented:
Just a couple of things:
1. Please revert the debug lines from CMakeLists.txt (line 6 & 16)
2. `newsitemwidget` should become `NewsItemWidget`
3. Please remove `Settings.ui`, as it has been removed here.
4. Please move `newspanel.ui` to a programmatically created UI
5. Please make sure that you are not using older versions of files (`MainPanel.cpp`)

DarvishKamalia committed:
- Removed debug flas from CMakeLists.txt
- Removed extraneous UI files

DarvishKamalia committed:
Fixed hover color

DarvishKamalia committed:
Fixed CMakeLists

DarvishKamalia committed:
Updated filenames in CMakeLists

DarvishKamalia committed:
Renamed NewsItemWidget files

thomassross commented on Source/Settings.cpp:
```diff
@@ -446,3 +452,8 @@ void Settings::confirmClearDb()
     }
 }
+
+void Settings::manageNewsFeed() {
```
This can be made into a lambda

DarvishKamalia committed:
- Changed connect to Qt5 syntax
- Changed some slots to lambda functions
